scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
+    { x = 145, y = 16, z = 427 },
+    { x = 173, y = 16, z = 458 },
+    { x = 178, y = 16, z = 460 },
+    { x = 210, y = 16, z = 460 },
+    { x = 215, y = 16, z = 457 },
+    { x = 219, y = 16, z = 452 },
+    { x = 219, y = 16, z = 442 },
+    { x = 219, y = 16, z = 420 },
+    { x = 240, y = 16, z = 419 },
+    { x = 258, y = 12, z = 417 },
+    { x = 259, y = 4, z = 384 },
+    { x = 254, y = 3, z = 380 },
+    { x = 239, y = 0, z = 379 },
+    { x = 214, y = 0, z = 380 },
+    { x = 208, y = 0, z = 386 },
+    { x = 208, y = 0, z = 398 },
+    { x = 206, y = 0, z = 405 },
+    { x = 198, y = 0, z = 409 },
+    { x = 191, y = 0, z = 405 },
+    { x = 192, y = 0, z = 395 },
+    { x = 193, y = 0, z = 389 },
+    { x = 187, y = 0, z = 382 },
+    { x = 184, y = 0, z = 380 },
+    { x = 164, y = 0, z = 380 },
 }
 
+    -- path from the first yellow door to the middle yellow door
+local firstYellowToMiddleYellow =
+{
+    { x = 156, y = 0, z = 379 },-- first yellow door
+    { x = 144, y = 0, z = 379},
+    { x = 133, y = 0, z = 380 },
+    { x = 129, y = 0, z = 384 },
+    { x = 124, y = 0, z = 399 },
+    { x = 124, y = 0, z = 410 },
+    { x = 117, y = 0, z = 424 },
+    { x = 114, y = 0, z = 439 },
+    { x = 111, y = 0, z = 451 },
+    { x = 109, y = 0, z = 456 },
+    { x = 106, y = 0, z = 458 },
+    { x = 100, y = 0, z = 460 },
+    { x = 94, y = 0, z = 459 },
+    { x = 82, y = 0, z = 459 },-- Middle yellow door
+}
+
+    -- path from the yellow middle door to the middle blue door
+local middleYellowToMiddleBlue =
+{
+    { x = 77, y = 0, z = 460 },
+    { x = 71, y = 0, z = 459 },
+    { x = 66, y = 0, z = 458 },
+    { x = 63, y = 0, z = 456 },
+    { x = 60, y = 0, z = 450 },
+    { x = 60, y = 0, z = 443 },
+    { x = 60, y = 0, z = 419 },
+    { x = 40, y = 0, z = 419 },
+    { x = 21, y = 0, z = 419 },
+    { x = -9, y = 0, z = 420 },
+    { x = -59, y = 0, z = 420 },
+    { x = -60, y = 0, z = 433 },
+    { x = -60, y = 0, z = 450 },
+    { x = -62, y = 0, z = 455 },
+    { x = -67, y = 0, z = 458 },
+    { x = -75, y = 0, z = 460 },
+    { x = -76, y = 0, z = 460 },
+}
+
+    -- Path from the middle blue door to the first blue door
+local middleBlueToFirstBlue =
+{
+    { x = -82, y = 0, z = 460 },
+    { x = -93, y = 0, z = 459 },
+    { x = -103, y = 0, z = 459 },
+    { x = -106, y = 0, z = 458 },
+    { x = -110, y = 0, z = 454 },
+    { x = -116, y = 0, z = 434 },
+    { x = -117, y = 0, z = 420 },
+    { x = -121, y = 0, z = 409 },
+    { x = -125, y = 0, z = 395 },
+    { x = -129, y = 0, z = 385 },
+    { x = -133, y = 0, z = 381 },
+    { x = -141, y = 0, z = 380 },
+    { x = -153, y = 0, z = 379 },
+    { x = -157, y = 0, z = 379 },-- Middle blue door
+
+}
 entity.onMobSpawn = function(mob)
+    local pos = mob:getPos()
     mob:setMod(xi.mod.REGAIN, 200)
-    mob:pathThrough(pathNodes, xi.path.flag.PATROL)
+
+    -- Zip popped on blue short side (!pos -196, 0, 396)
+    if pos.x <= -195 then
```

Comment:
It would be more obvious if you specify the spawn points as x/y/z tables, with a comment saying this is what's listed in the nm spawn points SQL, and then do `checkDistance()` on each of them

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
This has been extracted into a table at the start of the file.